docs/TESTCONTAINER_PLAN.md

```
@@ -0,0 +1,212 @@
+2. The `dist` target is a no-op if sources haven't changed (Make dependency tracking)
+3. It eliminates the fragility of shelling out to `make` from inside `go test`
+
+### No build tags
+
+No build tags or separate test targets. Any test that calls `testhelper.Setup(t)` will use the container. The `TestMain` Docker check ensures graceful degradation when Docker is unavailable.
+
+CI already runs tests in the `plugin-ci` reusable workflow which has Docker available on the runner.
+
+## Dependencies to add to go.mod
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)